docs/guide/overview.asciidoc

```diff
@@ -38,6 +36,42 @@ This data is buffered for a short period and sent on to APM Server.
 It listens on port 8200 by default and receives data from agents through a JSON HTTP API.
 Then it creates documents from that data and stores them in Elasticsearch.
 
-To visualize the data after it's sent to Elasticsearch,
-you can use the the dedicated APM UI bundled in X-Pack,
-or the pre-built open source Kibana dashboards that can be loaded directly via the {kibana-ref}/apm-getting-started.html[APM Kibana UI].
\ No newline at end of file
+*Elasticsearch* is a highly scalable open-source full-text search and analytics engine.
+It allows you to store, search, and analyze big volumes of data quickly and in near real time.
+
+*Kibana* is an open source analytics and visualization platform designed to work with Elasticsearch.
+You use Kibana to search, view, and interact with data stored in Elasticsearch indices.
+You an use Kibana to visualize APM data by utilizing the dedicated APM UI bundled in Basic license,
+or the pre-built, open source,
+Kibana dashboards that can be loaded directly via the {kibana-ref}/apm-getting-started.html[APM Kibana UI].
+
+[[concepts]]
+[float]
+=== APM Concepts
+
+*Agents*
+APM Agents capture different types of information from within their instrumented applications, known as events. Events can be <<transactions,`Transactions`>>, <<transaction-spans,`Spans`>>, and <<errors,`Errors`>>.
+These events are then streamed to a single endpoint in the APM Server which sorts and processes the events. 
+
+* <<transactions,`Transactions`>> - A transaction describes an event captured by an Elastic APM agent instrumenting a service. For example: an HTTP request, or a background job.
+* <<transaction-spans,`Spans`>> - Transactions can have 0, 1, or many spans. A span contains information about a specific code path that has been executed as part of a transaction.
+* <<errors,`Errors`>> - An error event contains information about the original exception that occurred or about a log created when the exception occurred.
+
+*Server*
+The APM Server is a {apm-server-ref}/why-separate-component.html[separate component by design] - it helps keep the agents light, prevents certain security risks, and improves compatibility across the Elastic and APM Stack.  
+
+The {apm-server-ref}/intake-api.html[Intake API] is what we call the internal protocol that APM agents use to talk to the APM Server. 
+After the APM Server has sorted and processed events from the APM agents (via the intake API), the server transforms the data into Elasticsearch documents and stores them in corresponding {apm-server-ref}/exploring-es-data.html[Elasticsearch indices]. In a matter of minutes you can start viewing your application performance data in the {kibana-ref}/apm-getting-started.html[Kibana UI] using our prebuilt dashboards.
```

Comment:
Is it on purpose that you do not mention the dedicated APM UI here, but only the prebuilt dashboards?

Reply:
It's mentioned earlier, but I need to better clarify that here as well. Thanks